Exercises/NoModules/Chapter 03/Soln3_01.cpp

```diff
@@ -0,0 +1,30 @@
+// Exercise 3-1. Output integer and its 1's complement in decimal and hexadecimal.
+
+// This tests how well you remember string formatting using std::format()
+// (see Chapter 2 if you forgot some of the formatting options), 
+// as well as two�s complement binary encoding and bitwise ~.
+
+#include <iostream>
+#include <format>
+
+// See Appendix A (available online) for static_assert().
+// You should adjust the field widths of the table if ints have a different size.
+static_assert(sizeof(int) == 4, "This program assumes 32 bit ints");
+
+int main()
+{
+  int value {};
+  std::cout << "Enter any integer: ";
+  std::cin >> value;
+  
+  const auto inverted = static_cast<unsigned>(~value);
+
+  // Output column headings (0b and 32 bits make for a width of 34)
+  std::cout << std::format(" {:^34} {:^34} {:^34}\n", "value", "~value", "~value + 1");
+
+  // Output binary values (cast to unsigned integers first for more realistic output)
+  std::cout << std::format(" {:#034b} {:#034b} {:#034b}\n", value, inverted, inverted + 1);
+
+  // Output decimal values
+  std::cout << std::format(" {:^34} {:^34} {:^34}\n", value, ~value, ~value + 1);
```

Comment:
the text says "each in hexadecimal representation and on one line", but this code outputs binary.
Also, the text says "where the values in the same column are right aligned", but you are centering in this solution;

Reply:
I had them change the exercise in the text:

"Create a program that prompts for input of an integer and store it as an int.
Invert all the bits in this value and store the result as an int. Next, convert this inverted value
to an unsigned int, and store it. Output the original value, the unsigned value with the bits
inverted, and that same inverted value plus 1, each in binary representation and on one line. On
the next line, output the original value, the signed inverted value, and the signed inverted value
plus 1, each in decimal representation. These two lines should be formatted such that they look
like a table, where the values in the same column are centered. You can add column headers
as well if you like. All binary values should have leading zeros so they are all 32 bits long
(assuming int values are 32 bits, which they usually are)."